tests/integration/coordinator/plugins/clickhouse/test_plugin.py

```diff
@@ -316,21 +316,32 @@ async def setup_cluster_content(clients: Sequence[HttpClickHouseClient], use_nam
     await clients[0].execute(b"INSERT INTO default.memory VALUES (123, 'foo')")
     await clients[0].execute(b"CREATE FUNCTION `linear_equation_\x80` AS (x, k, b) -> k*x + b")
     if await is_engine_available(clients[0], TableEngine.KeeperMap):
-        await clients[0].execute(
-            b"CREATE TABLE default.keeper_map (key UInt32, value String) ENGINE = KeeperMap('keeper_map') PRIMARY KEY key"
-        )
-        await clients[0].execute(b"INSERT INTO default.keeper_map VALUES (1, 'one'), (2, 'two')")
+        await add_keeper_map_tables(clients)
 
-        # wait for every zookeeper node to receive the update
-        async def keeper_map_has_replicated(client: ClickHouseClient) -> bool:
-            return len(await client.execute(b"SELECT * FROM default.keeper_map")) == 2
 
-        await wait_for_condition_on_every_node(
-            clients,
-            keeper_map_has_replicated,
-            "waiting for keeper_map to be replicated",
-            timeout_seconds=5,
-        )
+async def add_keeper_map_tables(clients: Sequence[ClickHouseClient]) -> None:
+    # add and drop a table with KeeperMap engine.  This leaves some metadata in ZooKeeper that we want to ignore.
+    await clients[0].execute(
+        b"CREATE TABLE default.keeper_map_dropped(key UInt32, value String) "
+        b"ENGINE = KeeperMap('keeper_map_dropped') PRIMARY KEY key"
+    )
+    await clients[0].execute(b"DROP TABLE default.keeper_map_dropped")
```

Comment:
This test checks if deleted table does not cause problems, but it does not tell if the new `try except` kicks it there. I will try to delete keeper map table and see what is in the zookeeper. But It would be even better if there will be a test showing that exactly those lines are executed when the table was deleted. (Because there may be intersection with other cases, like empty table, or deleted table deletes metadata and the issue is not fixed)

Reply:
This modified test fails on main, indicating that the added code is what kicks in.

You can see [here](https://github.com/ClickHouse/ClickHouse/blob/24.3/src/Storages/StorageKeeperMap.cpp#L566) that all zk paths except the parent path `zk_root_path` are dropped.

Reply:
I will add an empty table as a test case as well for completeness :)

Reply:
Thanks. I have also double checked it